### DISCUSSION TITLE
Group with FlatBox does not draw over children, but Pack does

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#775] [Q&A]
I know that the prescribed method of giving your widget group a background color is to set it's frametype to ```FlatBox```.  This works for ```Group```, but when I try it with ```Pack```, the child widget does not show up in front of the background.

```rust
use fltk::{
    app::*, enums::*, frame::*, group::*, prelude::*, window::*,
}; //fltk = { version = "1.0.22"}

fn main(){
    let app = App::default();
    let mut win = Window::new(200, 200, 500, 300, "Pack vs Group Overdraw");
    //child not visible
    let mut my_pack= Pack::new(50,50,100,200,"mypack");
    my_pack.set_frame(FrameType::FlatBox);
    my_pack.set_color(Color::Blue);
    //should see a yellow frame
    let mut frame = Frame::new(250,100,100,100, "Test");
    frame.set_frame(FrameType::FlatBox);
    frame.set_color(Color::Yellow);
    my_pack.end();

    //child shows up ontop of group correctly
    let mut my_group= Group::new(200,50,100,200,"my group");
    my_group.set_frame(FrameType::FlatBox);
    my_group.set_color(Color::Red);

    let mut frame = Frame::new(200,100,100,100, "Test");
    frame.set_frame(FrameType::FlatBox);
    frame.set_color(Color::Yellow);
    my_group.end();

    win.end();
    win.show();

    app.run().unwrap();
}
```
![image](https://user-images.githubusercontent.com/38958118/125670548-8e878f87-4e9d-4a14-b462-400404f145ba.png)

**Comment by wyhinton** [comment]
Turns out we're really not even supposed to be using ```Pack``` outside of a ```ScrollGroup```...🙃... See [this conversation ](https://groups.google.com/g/fltkgeneral/c/DaURdgoIpjo) on the FLTK General thread. So frustrating that FLTK doesn't have a built in FlexBox type Group container that just works as you might expect. 

As a side note, check out this [FL_Flex project](https://github.com/osen/FL_Flex). Wouldn't it be cool to have something like that in fltk-rs?

**Reply by MoAlyousef** [reply]
I probably wouldn’t say that about packs, they have limitations but they’re quite usable outside of scrolls, especially when you don’t need a scroll. 
As for Fl_Flex, the implementation seems easy enough to port to Rust. I’ll have a go at it some time soon.

**Reply by MoAlyousef** [reply]
It seems Fl_Flex is going to be added to FLTK:
https://groups.google.com/g/fltkcoredev/c/FiCv6o2lDmc

**Reply by MoAlyousef** [reply]
I have added a Flex widget based on the code in FL_Flex in master. The method names have been changed to snake case, and you would need to call set_type(FlexType::Row) (it defaults to Column), otherwise the api is fairly similar.

**Reply by MoAlyousef** [reply]
I have moved the Flex widget to its own crate (fltk-flex) and yanked 1.1.1. Trying to port the examples I had run into several issues which I’ll need time to debug, mainly adding a child Flex into a parent Flex requires the child to have an initial size.

**Comment by mvasi90** [comment]
@wyhinton, hello.

The FL_Pack is not made to be used by itself. The FLTK documentation is incomplete.
The FL_Pack should be used in a FL_Scroll (but also works in a FL_Group).

To solve this problem, wrap the `group::Pack` in a `group::Group`, and apply the frame type and color to the group instead the pack:
```rust
...
let mut my_group = Group::new(50,50,100,200,None);
my_group.set_frame(FrameType::FlatBox);
my_group.set_color(Color::Blue);
let mut my_pack= ...;
...
```
